R/modules.R

```diff
@@ -103,6 +103,36 @@ append_module <- function(modules, module) {
   modules
 }
 
+#' Extract a `teal_module_landing` from list of `modules`
+#' @param modules `teal_modules`
+#' @keywords internal
+#' @return `teal_module_landing`
+extract_landing <- function(modules) {
+  if (inherits(modules, "teal_module_landing")) {
+    modules
+  } else if (inherits(modules, "teal_module")) {
+    NULL
+  } else if (inherits(modules, "teal_modules")) {
+    Filter(function(x) length(x) > 0L, lapply(modules$children, extract_landing))
+  }
+}
+#' Remove a `teal_module_landing` from list of `modules`
+#' @param modules `teal_modules`
+#' @keywords internal
+#' @return `teal_modules`
+drop_landing <- function(modules) {
+  if (inherits(modules, "teal_module_landing")) {
+    NULL
+  } else if (inherits(modules, "teal_module")) {
+    modules
+  } else if (inherits(modules, "teal_modules")) {
+    do.call(
+      "modules",
```

Comment:
can we avoid calling `modules()` please?
more info: https://github.com/insightsengineering/teal/issues/889

Reply:
:monocle_face: 
There is this line in `init`:
```
  if (inherits(modules, "list")) {
    modules <- do.call(teal::modules, modules)
  }
```
So internally the `modules` argument` is a `teal_modules`, a result of `modules`. I don't think the deprecation mentioned in #889 has properly started yet.

Reply:
yeah @pawelru but there is no clear substitution for `modules()` yet. if we come up with a solution for deprecation of `modules()` then we can easily remove it here. In this implementation we get back to `modules()` structure which is still supported and the other/next part of `teal::init` assumes `modules` structure is passed.